packages/owlbot-bootstrapper/common-container/fetch-api-info.ts

```diff
@@ -0,0 +1,211 @@
+  let apis;
+  if (contents) {
+    apis = JSON.parse(contents.toString()).apis;
+  }
+
+  return extractApiInfoFromJson(apis, apiId);
+}
+/**
+ * Function that extracts information for a given API from apis.json
+ *
+ * @param apis the apis.json file downloaded from drift as a JSON object
+ * @param apiId the unique identifier of the API
+ * @returns DriftAPI the fields as described in Drift
+ */
+function extractApiInfoFromJson(apis: DriftApi[], apiId: string): DriftApi {
+  const shortName = apiId.split('.')[apiId.split('.').length - 2];
+  for (const api of apis) {
+    if (
+      api.api_shortname === shortName ||
+      api.api_shortname.includes(`cloud${shortName}`)
+    ) {
+      return api;
+    }
+  }
+
+  // If we don't find a match, return an empty object
+  logger.info(
+    `There was no match in apis.json for ${apiId}, shortname: ${shortName}`
+  );
+  return {
+    api_shortname: shortName,
+    display_name: '',
+    docs_root_url: '',
+    launch_stage: 'LAUNCH_STAGE_UNSPECIFIED' as ReleaseLevel,
+    github_label: '',
+  };
+}
+
+function transformApiIdToPath(apiId: string): string {
+  return apiId.toString().replace(/\./g, '/');
+}
+
+/**
+ * Function that gets information from the service_config.yaml file in googleapis/googleapis
+ *
+ * @param apiId unique identifier fo API ID
+ * @param repositoryFileCache wrapper for getting repository contents from github (allows globbing)
+ * @returns a yaml object
+ */
+async function getApiProtoInformation(
+  apiPath: string,
+  repositoryFileCache: RepositoryFileCache
+): Promise<Partial<ServiceConfigYaml>> {
+  let yamlFilePath;
+  let yamlFile;
+  try {
+    yamlFilePath = (
+      await repositoryFileCache.findFilesByGlob('*_v*.yaml', 'main', apiPath)
+    )[0];
+  } catch (e) {
+    if (e instanceof FileNotFoundError) {
+      logger.warn('service_config.yaml path was not found');
+    } else {
+      // rethrow
+      throw e;
+    }
+  }
+
+  try {
+    if (yamlFilePath) {
+      yamlFile = await repositoryFileCache.getFileContents(
+        yamlFilePath,
+        'main'
+      );
+    } else {
+      logger.warn('service_config.yaml path was not found');
+    }
+  } catch (e) {
+    if (e instanceof FileNotFoundError) {
+      logger.warn('service_config.yaml was not found');
+    } else {
+      throw e;
+    }
+  }
+
+  let parsedYaml;
+  if (yamlFile && yamlFile.parsedContent) {
+    parsedYaml = yaml.load(yamlFile.parsedContent);
+  }
+
+  if (!parsedYaml) {
+    throw new Error('Service config not valid yaml or undefined');
+  }
+
+  return parsedYaml;
+}
+
+function assignDriftValuesToServiceConfig(
+  serviceConfig: Partial<ServiceConfigYaml>,
+  driftData: DriftApi
+): ServiceConfigYaml {
+  if (!serviceConfig?.api_short_name) {
+    logger.info(
+      `Service config did not contain api_short_name; replacing with DRIFT ${driftData.api_shortname}`
+    );
+    serviceConfig.api_short_name = driftData.api_shortname;
+  }
+
+  if (!serviceConfig?.github_label) {
```

Comment:
In my understanding, `github_label`, `api_shortname`, `documentation_uri`, `doc_tag_prefix` are under `publishing`, not at the top level.

Reply:
Updated accordingly; thank you!